activerecord/lib/active_record/relation.rb

```diff
@@ -642,14 +650,36 @@ def delete_by(*args)
       where(*args).delete_all
     end
 
+    # Schedule the query to be performed from a background thread pool.
+    #
+    #   Post.where(published: true).load_async # => #<ActiveRecord::Relation>
+    def load_async
+      unless loaded?
```

Comment:
The use of guard clause will simplify most of the conditional statements in the codebase.

Reply:
Thank you for the suggestion but we prefer to not use guard clauses.

Reply:
Is there a place where the core team explains why they don't want to use guard clauses? I did a quick search, but wasn't able to find anything and would love to hear the reasons.